### PR TITLE
ENH: ReduceLROnPlateau records the learning rate and works on batches

### PR DESCRIPTION
**Pull request by BenjaminBossan**
Previously, when using `ReduceLROnPlateau`, we would not record the learning rates in history. The code comment says that's because this class does not expose the `get_last_lr` method. I checked it again and it's now present, so let's use it.

Furthermore, I made a change to enable `ReduceLROnPlateau` to step on each batch (so far, only each epoch was supported). This is consistent with other learning rate schedulers.

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
The CI is failing because of a change in sklearn 1.6 and requires PR #1076.

**Review by githubnemo — APPROVED** [review]
